### PR TITLE
Ignore parameter values in find_hung_process.py

### PR DESCRIPTION
**Pull request by permcody**
Adding new functionality to `find_hung_process.py`

refs #25870

### CODE REVIEW

File: framework/scripts/find_hung_process.py [hunk 1]
```diff
@@ -501,6 +507,8 @@ def main():
     parser.add_argument('-s', '--stacks', metavar='int', action='store', type=int, default=0,
                         help='The number of stack frames to keep and compare for '
                         'uniqueness (Default: ALL)')
+    parser.add_argument('-p', '--param-values', action='store_true', dest="params_are_significant", default=False,
+                        help="Params are significant when comparing stack strings (Default: False)")
```
Comment (GiudGiud) [inline framework/scripts/find_hung_process.py:511]: change to:
"Parameters which are significant ........................... (Default: all significant)"

File: framework/scripts/find_hung_process.py [hunk 2]
```diff
@@ -331,6 +331,7 @@ def __init__(self):
         self.host_and_frames_re = re.compile('^(Host.*?)\n(.*)', re.M | re.S)
         self.memory_re = re.compile(r'0x[0-9a-f]*')    # Pattern to match (and remove) memory addresses
         self.frame_num_re = re.compile(r'^#\d.', re.M) # frame numbers IMPORTANT: Always match 2 characters due to fixed width formatting!
+        self.param_val_re = re.compile(r'(\w+?)=\S+(?=(?:,|\)))') # Pattern to match (and ignore) specific param values
```
Comment (GiudGiud) [inline framework/scripts/find_hung_process.py:334]:
```suggestion
        self.param_val_re = re.compile(r'(\w+?)=\S+(?=(?:,|\)))') # Pattern to match (and ignore the others) specific param values
```

File: framework/scripts/find_hung_process.py (hunk 1 above)
Comment (GiudGiud) [reply framework/scripts/find_hung_process.py:511]: what do you mean by parameter?

File: framework/scripts/find_hung_process.py (hunk 2 above)
Comment (permcody) [reply framework/scripts/find_hung_process.py:334]: No, this is not what I want to say. If anything dropping the parentheses would be the only thing that makes sense. I just want to ignore parameter values.

File: framework/scripts/find_hung_process.py (hunk 1 above)
Comment (permcody) [reply framework/scripts/find_hung_process.py:511]: > 13 0x000015202aea0648 in libMesh::PetscVector<double>::init (this=this@entry=foo, n=437434470, n_local=n_local@entry=970200, ghost=..., fast=fast@entry=false, ptype=ptype@entry=libMesh::GHOSTED) at /apps/local/spack/opt/spack/gcc-8.4.1/gcc-9.4.0-57pg55xkr4cydqecxijtmaaptbrgkxcn/include/c++/9.4.0/ext/new_allocator.h:89

See the line above, what I really mean is parameter values are significant (the lhs of those name/value pairs).

Comment (GiudGiud) [reply framework/scripts/find_hung_process.py]: so 
```
Set whether the argument values for each routine/level in the stack trace are significant for the comparison of stack traces between ranks
```

File: framework/scripts/find_hung_process.py (hunk 2 above)
Comment (GiudGiud) [reply framework/scripts/find_hung_process.py:334]: alright then
```
Pattern to match and ignore parameter values for a specific parameter of any routine
```

Comment (permcody) [reply framework/scripts/find_hung_process.py:334]: Not a specific parameter value, we either pay attention to all of them, or none of them.

Comment (GiudGiud) [reply framework/scripts/find_hung_process.py:334]: then dont say specific ?
```
Pattern to match and ignore parameter values for all parameters of any routine
```

File: framework/scripts/find_hung_process.py [hunk 3]
```diff
@@ -474,6 +475,11 @@ def compare_traces(self, trace1, trace2):
         trace1 = self.frame_num_re.sub('#00', trace1)
         trace2 = self.frame_num_re.sub('#00', trace2)
 
+        # Drop specific param values
```
Comment (GiudGiud) [inline framework/scripts/find_hung_process.py:478]:
```suggestion
        # Drop parameter values from comparison
```

File: framework/scripts/find_hung_process.py [hunk 4]
```diff
@@ -362,7 +363,7 @@ def __longest_trace(self):
         """
         return max(trace.count('\n') for trace in self.traces)
 
-    def process_traces(self, num_lines_to_keep):
+    def process_traces(self, num_lines_to_keep, significant_params):
```
Comment (GiudGiud) [inline framework/scripts/find_hung_process.py:366]: params_are_significant is a better name, let s keep that one

Comment (GiudGiud) [reply framework/scripts/find_hung_process.py:366]: what s up with that one?
```suggestion
    def process_traces(self, num_lines_to_keep, params_are_significant):
```
significant_params would be for the list of significant parameters

**Review by GiudGiud — APPROVED** [review]

**Comment by permcody** [comment]
Blackbear doc build is unrelated.

**Pull request merged**